pkg/kv/kvserver/store_create_replica.go

```diff
@@ -252,12 +252,12 @@ func fromReplicaIsTooOldRLocked(toReplica *Replica, fromReplica *roachpb.Replica
 	return !found && fromReplica.ReplicaID < desc.NextReplicaID
 }
 
-// addToReplicasByKeyLockedReplicaRLocked adds the replica to the replicasByKey
-// btree. The replica must already be in replicasByRangeID. Returns an error if
-// a different replica with the same range ID, or an overlapping replica or
-// placeholder exists in this Store. Replica.mu must be at least read-locked.
-func (s *Store) addToReplicasByKeyLockedReplicaRLocked(repl *Replica) error {
-	desc := repl.descRLocked()
+// addToReplicasByKeyLocked adds the replica to the replicasByKey btree. The
+// replica must already be in replicasByRangeID. Returns an error if a different
+// replica with the same range ID, or an overlapping replica or placeholder
+// exists in this Store.
+func (s *Store) addToReplicasByKeyLocked(repl *Replica) error {
+	desc := repl.Desc()
```

Comment:
Now this line deadlocks through `applySnapshot` -> `r.mu.Lock()` -> `markReplicaInitializedLockedReplLocked` -> `addToReplicasByKeyLocked` -> `repl.Desc()` -> `repl.mu.RLock()`.

Reply:
Fixed.